soc/arm/atmel_sam0/samd21/Kconfig.defconfig.series

```diff
@@ -30,4 +30,11 @@ config NUM_IRQS
 config SYS_CLOCK_HW_CYCLES_PER_SEC
 	default 48000000
 
+if BUILD_OUTPUT_UF2
+
+config BUILD_OUTPUT_UF2_FAMILY_ID
+	default "0x68ed2b88"
+
+endif
+
```

Comment:
Retargeting the discussion from https://github.com/zephyrproject-rtos/zephyr/pull/31066#discussion_r576094245: on further review I now completely disagree with the suggestion to move these out to a single list maintained separately.

The UF2 IDs appear to come [from the uf2 script](https://github.com/microsoft/uf2/blob/6e0f282d4b1a56548e4780de95498e9de158fe37/utils/uf2conv.py#L15) and are associated with specific SoC families.

So I think these belong exactly here, in the SoC Kconfig, and that the value for `BUILD_OUTPUT_UF2_FAMILY_ID` should be the text identifier for the family, in this case `"SAMD21"`, rather than the hex constant.

Then when the script is updated to support more SoCs the corresponding SoC Kconfigs should have the property added.

Also since there's only one default here it seems cleaner to use:

```
config BUILD_OUTPUT_UF2_FAMILY_ID
	depends on BUILD_OUTPUT_UF2
	default "SAMD21"
```

rather than an enclosing conditional, especially since that is the [underlying equivalent](https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#factoring-out-common-dependencies).

Reply:
> Retargeting the discussion from [#31066 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/31066#discussion_r576094245): on further review I now completely disagree with the suggestion to move these out to a single list maintained separately.
> 
> The UF2 IDs appear to come [from the uf2 script](https://github.com/microsoft/uf2/blob/6e0f282d4b1a56548e4780de95498e9de158fe37/utils/uf2conv.py#L15) and are associated with specific SoC families.

I disagree there.
This is no different than drivers Kconfig, Bluetooth, TF-M where it is no code owners responsible for those areas who will have defaults in their Kconfig files.
This also goes for boards.

All such knowledge doesn't go into boards or soc Kconfigs, as this will become unmaintainable.
https://github.com/zephyrproject-rtos/zephyr/blob/a890790592d85d7804f5059618d94e1acaeebe5a/drivers/flash/Kconfig.stm32#L7-L32

https://github.com/zephyrproject-rtos/zephyr/blob/a890790592d85d7804f5059618d94e1acaeebe5a/modules/Kconfig.tfm#L8-L16

and many more.

With your rationale, anything that has a default that is based on a specific SoC could then be placed in the corresponding Kconfig soc file.

A SoC maintainer should not need to consider UF2, this is for whoever maintains the UF2 feature, and I think it's UF2 is to be seen as independent feature.

Reply:
OK, that's somewhat compelling, but then the Kconfig defaults should go into the board Kconfig.defconfig for boards that are likely to support UF2 bootloaders, just as Nordic default-enables `BT_CTLR` for their boards.

This isn't using a module, so its configuration doesn't belong in `modules/`.  Either put it as defaults where the variable is defined, or put it in the board Kconfig.defconfig.  Hiding it elsewhere just makes it painful.